### PR TITLE
T3T1 ShareWords component

### PR DESCRIPTION
**Pull request by obrusvit**
Known limitations at the moment

- the footer (with instructions) is simplified and does not contain "i of 12" counter
- title is all caps until we fix https://github.com/trezor/trezor-firmware/issues/3600 
- reacts to `Swipe` (up/down), must be revisited when we progress with swipe navigation

recovery_9_office

### CODE REVIEW

File: core/embed/rust/src/ui/model_mercury/layout.rs [hunk 1]
```diff
@@ -2082,9 +2081,9 @@ pub static mp_module_trezorui2: Module = obj_module! {
     /// def show_share_words(
     ///     *,
     ///     title: str,
-    ///     pages: Iterable[str],
+    ///     share_words: Iterable[str],
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/layout.rs:2084]: Breaks style check: https://github.com/trezor/trezor-firmware/actions/runs/8491733359/job/23263990332?pr=3656

Probably best to name the function something new, until we have modtrezorui-per-model.

File: core/embed/rust/src/ui/model_mercury/component/share_words.rs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    }
+}
+
+impl<'a> Component for ShareWords<'a> {
+    type Msg = PageMsg<()>;
+
+    fn place(&mut self, bounds: Rect) -> Rect {
+        self.area = bounds;
+        let (_, bot) = bounds.split_top(33);
+        let area_word = bot.split_top(91).0;
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/component/share_words.rs:54]: Please make these named constants.

**Review by mmilata — COMMENTED** [review]
Looking good. Have you tried running device tests with the changes?

File: core/embed/rust/src/ui/model_mercury/layout.rs (hunk 1 above)
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/layout.rs]: Fixed here. Also the device tests for reset are running. I haven't updated fixtures though.
5b50beaa5de417834f97062b346aa1926f2fe120

File: core/embed/rust/src/ui/model_mercury/component/share_words.rs (hunk 2 above)
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/component/share_words.rs]: a7c366da26755159298ed2e60dced2316326f151

**Comment by obrusvit** [comment]
This is what it looks like with offsets @Hannsek 
emu00000000

File: core/embed/rust/src/ui/model_mercury/constant.rs [hunk 3]
```diff
@@ -20,3 +20,8 @@ pub const fn screen() -> Rect {
     Rect::from_top_left_and_size(Point::zero(), SIZE)
 }
 pub const SCREEN: Rect = screen();
+
+/// Spacing between components (e.g. header and main content) and offsets from
+/// the side of the screen. Generally applied everywhere except the top side of
+/// the header. [px]
+pub const SPACING: i16 = 2;
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/constant.rs:27]: Nit: probably more of a `theme.rs` concern but not a huge difference

**Review by mmilata — APPROVED** [review]
Thanks for fixing the tests, feel free to merge.

**Pull request merged**